lib/src/auth/user_gateway.dart

```diff
@@ -5,9 +5,13 @@ import 'package:firedart/src/auth/token_provider.dart';
 
 class UserGateway {
   final UserClient _client;
+  final String authGatewayUrl;
 
-  UserGateway(KeyClient client, TokenProvider tokenProvider)
-      : _client = UserClient(client, tokenProvider);
+  UserGateway(
```

Comment:
I believe we need to add documentation to the parts we are changing. Please take a look at other constructors we modified in this PR.